src/Orchard.Environment.Extensions/ExtensionManager.cs

```diff
@@ -256,6 +256,11 @@ public Task<ExtensionEntry> LoadExtensionAsync(IExtensionInfo extensionInfo)
 
         public async Task<IEnumerable<FeatureEntry>> LoadFeaturesAsync()
         {
+            // for testing
+            Parallel.ForEach(GetFeatures(),
```

Comment:
Oh, no more on this branch, i will check it again.

So, it was to compare `Parallel.ForEach` vs `Task.WhenAll()` when using `LoadFeatureAsync()` which the 1st time triggers extension compilation / loading.

So the idea was to use full parallelism with the 1st while `Task.WhenAll()` create only one task which wait on all async Task<TResult> objects. But because our extension compilation / loading are not called through async methods, i think these parts are executed "synchronously".

But quickly done without really checking this part of code.

So **based on your comment where you say it doesn't wait** i will return back to this branch and test it again.